code/src/hooks/use-filters-with-apply/useFiltersWithApply.test.ts

```diff
@@ -0,0 +1,160 @@
+import { act, renderHook } from "@testing-library/react";
+import { useSearchParams } from "react-router-dom";
+
+import parseFiltersFromParams from "@/hooks/use-filters-with-apply/parse-filters-from-params/parseFiltersFromParams";
+import serializeToQueryString from "@/hooks/use-filters-with-apply/serialize-to-query-string/serializeToQueryString";
+import useFiltersWithApply from "@/hooks/use-filters-with-apply/useFiltersWithApply";
+
+jest.mock("react-router-dom", () => ({
+  useSearchParams: jest.fn()
+}));
+
+jest.mock(
+  "@/hooks/use-filters-with-apply/parse-filters-from-params/parseFiltersFromParams"
+);
+jest.mock(
+  "@/hooks/use-filters-with-apply/serialize-to-query-string/serializeToQueryString"
+);
+
+const mockSetSearchParams = jest.fn();
+
+describe("useFiltersWithApply", () => {
+  let defaultFilters: Record<string, unknown>;
+
+  beforeEach(() => {
+    defaultFilters = { filter1: "default1", filter2: "default2" };
+
+    (useSearchParams as jest.Mock).mockReturnValue([
+      new URLSearchParams(),
+      mockSetSearchParams
+    ]);
+
+    (parseFiltersFromParams as jest.Mock).mockReturnValue({
+      defaultActiveFilters: new Set(),
+      defaultFiltersFromParams: defaultFilters
+    });
+
+    (serializeToQueryString as jest.Mock).mockImplementation((value: object) =>
+      value.toString()
+    );
```

Comment:
move `mocks` above

Reply:
Used your suggested approach, thank you Dmytro